File: app_Web/src/interface/entity/ibizpro-index.ts
/**
 * 索引检索
 *
 * @export
 * @interface IbizproIndex
 */
export interface IbizproIndex {

    /**
     * 类型
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    indextype?: any;

    /**
     * 主键
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    indexid?: any;

    /**
     * 支持搜索[需求、Bug、任务、项目、产品、文档、用例]
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    indexname?: any;

    /**
     * 逻辑标识
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    deleted?: any;

    /**
     * 组织标识
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    orgid?: any;

    /**
     * 部门标识
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    mdeptid?: any;

    /**
     * 描述
     *
     * @returns {*}
     * @memberof IbizproIndex
     */
    indexdesc?: any;
}

File: ibizpms-core/src/main/java/cn/ibizlab/pms/core/es/domain/IbizproIndex.java
package cn.ibizlab.pms.core.es.domain;

import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;
import org.springframework.data.annotation.Id;
import java.io.Serializable;
import java.math.BigDecimal;
import java.sql.Timestamp;
import lombok.Data;

@Data
@Document(indexName = "ibizproindex", type = "_doc", shards = 5, replicas = 1)
public class IbizproIndex implements Serializable {

    private static final long serialVersionUID = 1L;

    @Field(type = FieldType.Long)
    private Long indexid;
    @Field(type = FieldType.Text)
    private String acl;
    @Id
    private String docid;
    @Field(type = FieldType.Text)
    private String indexname;
    @Field(type = FieldType.Text)
    private String color;
    @Field(type = FieldType.Text)
    private String deleted;
    @Field(type = FieldType.Text)
    private String acllist;
    @Field(type = FieldType.Text)
    private String indexdesc;
    @Field(type = FieldType.Long)
    private Long product;
    @Field(type = FieldType.Text)
    private String indextype;
    @Field(type = FieldType.Text)
    private String mdeptid;
    @Field(type = FieldType.Long)
    private Long project;
    @Field(type = FieldType.Text)
    private String orgid;
}





